open-window.csl

```diff
@@ -177,38 +146,104 @@
       </group>
       <group delimiter=" ">
-        <label variable="volume" form="short" text-case="capitalize-first"/>
+        <label text-case="capitalize-first" variable="volume"/>
         <text variable="volume"/>
       </group>
-      <text macro="editor-translator"/>
+      <group font-variant="normal">
+        <text macro="editor-translator" font-variant="normal"/>
+      </group>
+      <group delimiter=". ">
+        <choose>
+          <if type="broadcast" match="any">
+            <text variable="number" prefix="Episode "/>
+          </if>
+        </choose>
+        <text variable="version"/>
+        <choose>
+          <if type="broadcast" match="none">
+            <text variable="medium"/>
+          </if>
+        </choose>
+      </group>
       <group delimiter=" " prefix="(" suffix=")">
         <text variable="collection-title"/>
         <group delimiter=" ">
           <label variable="issue" form="short"/>
           <text variable="collection-number"/>
         </group>
       </group>
+      <choose>
+        <if type="paper-conference" match="any">
+          <date form="text" variable="issued"/>
+        </if>
+      </choose>
       <text macro="publisher"/>
     </group>
   </macro>
   <macro name="publisher">
-    <group delimiter=": ">
-      <text variable="publisher-place"/>
+    <group>
+      <text variable="publisher-place" suffix=": "/>
       <text variable="publisher"/>
+      <text variable="page" prefix=":"/>
     </group>
   </macro>
+  <macro name="access">
+    <choose>
+      <if type="book chapter thesis" match="none">
+        <choose>
+          <if variable="DOI">
+            <group delimiter=" ">
+              <group prefix="https://doi.org/">
+                <text variable="DOI"/>
+              </group>
+            </group>
+          </if>
+          <else-if variable="URL" match="all">
+            <group delimiter=" ">
+              <text variable="genre" suffix="."/>
+              <text variable="URL"/>
+              <text macro="date-access"/>
+            </group>
+          </else-if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+  <macro name="date-issue-long">
+    <choose>
+      <if match="any" variable="issued">
+        <date form="text" date-parts="year-month-day" variable="issued">
+          <date-part name="day" form="numeric-leading-zeros"/>
+          <date-part name="month"/>
+          <date-part name="year"/>
+        </date>
+      </if>
+    </choose>
+  </macro>
+  <macro name="author-count">
+    <names variable="author">
+      <name form="count"/>
+    </names>
+  </macro>
   <citation disambiguate-add-year-suffix="true" year-suffix-delimiter="," disambiguate-add-names="true" disambiguate-add-givenname="false" collapse="year-suffix">
     <sort>
       <key macro="date-issued-no-parentheses"/>
       <key macro="author"/>
     </sort>
     <layout prefix="(" suffix=")" delimiter="; ">
```

Comment:
And the delimiter here should be a comma I think

Reply:
The delimiter here should be a semicolon, this is correct.